docs/tutorials/portable-consent.md

```diff
@@ -0,0 +1,137 @@
+
+To ensure that your users’ consent settings are consistently applied in any app built with XMTP, contact consent states are stored and accessible to apps in a consent list on the XMTP network.
+
+For example, if your user blocked a contact on a different device or in another app and then uses your app, you can load the latest consent list to be sure to not mistakenly display conversations with the blocked contact.
+
+## Understand contact consent states
+
+A contact can have one of three consent states:
+
+- Unknown
+- Allowed
+- Blocked
+
+Here are some of the ways contact consent states are set:
+
+**Unknown**
+
+- For a new conversation that a new contact wants to start with your user, the contact consent state is set to `unknown`.
+    
+    For example, you can consider displaying a conversation with an `unknown` contact on a **Requests** tab and give your user the option to block or allow the contact.
+    
+- For a preexisting conversation with only messages from a contact that hasn’t had its consent state updated on the network, the contact consent state is set to `unknown` by default.
+    
+    For example, you can consider displaying a conversation with an `unknown` contact on a **Requests** tab and give your user the option to block or allow the contact.
+    
+**Allowed**
+
+- For a new conversation that your user requests with a contact, the contact consent state is set to `allowed`. Your user’s creation of the conversation with the contact is considered as consenting to the conversation with the contact.
+    
+    For example, you can consider displaying the conversation with the `allowed` contact on the **Messages** tab.
+    
+- For a preexisting conversation with one or more messages from a contact that hasn’t had its consent state updated on the network (`unknown`) and your user has taken the action to send a message in response, update the contact consent state to `allowed`. You can consider the response from your user as consenting to the conversation with the contact.
+    
+    For example, you can consider moving the conversation with the contact from the **Requests** tab to the **Messages** tab.
+    
+- For a contact that your user has taken the action to allow, update the contact consent state to `allowed`.
+
+    For example, you can consider displaying the conversation with the allowed contact on the **Messages** tab in your user’s inbox.
+
+**Blocked**
+
+- For a contact that your user has taken the action to block, update the contact consent state to `blocked`.
+    
+    For example, you can consider removing a conversation with the blocked contact from your user’s inbox completely.
+    
+## Build portable consent
+
+Use the following methods to build portable consent in your app.
+
+### Block or allow contacts
+
+To enable your user to block or allow a contact, call the following methods. Note that these functions accept lists, so you can do batch requests.
```

Comment:
Do we want to add code samples beyond TS?

Reply:
FWIW these code samples are React Native. I can provide samples for iOS and Andriod as well. @rygine could probably provide for the React, JS side.

Reply:
I've added placeholders for Fabri to provide the JS code samples. We can always publish and just keep adding code samples for more languages as they become available. Thank you so much!